test/metabase/models/setting_test.clj

```diff
@@ -1091,3 +1091,57 @@
         (catch Exception e
           (is (re-matches #"defsetting docstrings must be a \*deferred\* i18n form.*"
                           (:cause (Throwable->map e)))))))))
+
+(defsetting test-setting-audit-never
+ "Test setting with no auditing"
+  :audit :never)
+
+(defsetting test-setting-audit-raw-value
+  "Test setting with auditing raw values"
+  :type  :integer
+  :audit :raw-value)
+
+(defsetting test-setting-audit-getter
+  "Test setting with auditing values returned from getter"
+  :type :string
+  :getter (constantly "GETTER VALUE")
+  :audit :getter)
+
+(deftest setting-audit-test
+  (let [last-audit-event-fn #(t2/select-one [:model/AuditLog :topic :user_id :model :details]
+                                            :topic :setting-update
+                                            {:order-by [[:id :desc]]})]
+    (testing "Settings are audited by default without their value included"
+      (test-setting-1! "DON'T INCLUDE THIS VALUE")
+      (is (= {:topic   :setting-update
+              :user_id  nil
+              :model   "Setting"
+              :details {:key "test-setting-1"}}
+             (last-audit-event-fn))))
+
+    (testing "Auditing can be disabled with `:audit :never`"
+      (test-setting-audit-never! "DON'T AUDIT")
+      (is (not= "DON'T AUDIT"
```

Comment:
Wouldn't this be true for a setting that was audited by default (`:audit :no-value` vs `:audit :never`)? It'd have the `:details` key with `{:key "test-setting-audit-never!"}` value, I think.

Said another way: Would it make more sense to check that this is `not= "test-setting-audit-never!"`?

Reply:
Yep, good catch thanks